### PR TITLE
CLI: manage users

### PR DESCRIPTION
**Pull request by jburel**
Review admin CLI pages for Basel training

See 
* https://trello.com/c/4zhuts6g/28-docs-rfe-deactivate-user-cli
* https://trello.com/c/hutj8DLg/29-docs-omero-user-edit-via-cli
* https://trello.com/c/jHe6FDXW/31-cli-docs-password-in-env-var
* https://trello.com/c/s5flLgPw/32-cli-docs-sessions-list-to-be-updated-to-sessions-who

### CODE REVIEW

File: omero/sysadmins/cli/usergroup.rst [hunk 1]
```diff
@@ -67,6 +67,18 @@ Use the command :program:`omero user leavegroup` and specify the ``user`` group
     # Remove jsmith from group user
     $ bin/omero user leavegroup user --name=jsmith
 
+Edit user
+^^^^^^^^^
+
+Updating the details of a user e.g. the email address can be achieved using
+:program:`omero obj update` command:
```
Comment (mtbc) [inline omero/sysadmins/cli/usergroup.rst]: +"the"

File: omero/sysadmins/cli/usergroup.rst [hunk 2]
```diff
@@ -57,6 +57,28 @@ below:
...
+User deactivation
+^^^^^^^^^^^^^^^^^
+
+To deactivate a user, remove him/her from the system group ``user`` .
+Use the command :program:`omero user leavegroup` and specify the ``user`` group as the target::
+
+    # Remove jsmith from group user
+    $ bin/omero user leavegroup user --name=jsmith
+
+User edition
```
Comment (mtbc) [inline omero/sysadmins/cli/usergroup.rst]: editing?

File: omero/sysadmins/cli/usergroup.rst [hunk 3]
```diff
@@ -57,6 +57,28 @@ below:
...
+^^^^^^^^^^^^
+
+Updating the details of a user e.g. the email address can be achieved using
+the :program:`omero obj update` command:
+
+    # Determine the ID of jsmith
+    $ bin/omero user info --user-name jsmith
+    # Change the email address of jsmith
+    $ bin/omero obj update Experimenter:ID_jsmith email=jsmith@new_address.com
```
Comment (mtbc) [inline omero/sysadmins/cli/usergroup.rst]: not sure the ID number substitution is obvious here

File: omero/users/cli/chgrp.rst [hunk 4]
```diff
@@ -17,14 +17,14 @@ Who may move data
 * a full administrator
 * a :doc:`restricted administrator
   </sysadmins/restricted-admins>` with `Chgrp` privilege
-* the owner of the data *if* they are a member of the target group
+* the owner of the data *if* he/she is a member of the target group
```
Comment (mtbc) [inline omero/users/cli/chgrp.rst]: Don't see that this is needed -- singular "they" has a fine literary history -- but no strong objection.

File: omero/sysadmins/cli/usergroup.rst [hunk 5]
```diff
@@ -19,15 +19,15 @@ command::
 
 During the addition of the new user, you will need to specify the first and
 last name of the new user and their username as well as the groups the user
-belongs to. To add John Smith as a member of group 2 identified as jsmith,
+belongs to. To add John Smith as a member of the group named ``test-group`` identified as jsmith,
```
Comment (manics) [inline omero/sysadmins/cli/usergroup.rst]: Format jsmith: ``` ``jsmith`` ```

File: omero/sysadmins/cli/usergroup.rst [hunk 6]
```diff
@@ -57,6 +57,33 @@ below:
...
+To reactivate the user, add him/her back to the system group ``user`` i.e.::
+
+    $ bin/omero group adduser jsmith --name=user
+
+
+User editing
+^^^^^^^^^^^^
+
+Updating the details of a user e.g. the email address can be achieved using
+the :program:`omero obj update` command:
```
Comment (manics) [inline omero/sysadmins/cli/usergroup.rst]: I think you need a double `::` at the end

File: omero/users/cli/sessions.rst [hunk 7]
```diff
@@ -93,10 +95,10 @@ Some of the options available to the :program:`omero login` command are:
 Multiple sessions
 ^^^^^^^^^^^^^^^^^
 
-Stored sessions can be listed using the :program:`omero sessions list`
+Stored sessions can be listed using the :program:`omero sessions who`
```
Comment (manics) [inline omero/users/cli/sessions.rst]: `sessions list` looks more appropriate to me:
```
$ omero sessions who
Using session for user-5@eel.openmicroscopy.org:4064. Idle timeout: 10 min. Current group: read-write-1
 name   | group        | logged in               | agent          | timeout | hitCount | lastAccessTime
--------+--------------+-------------------------+----------------+---------+----------+---------------------
 user-3 | read-only-1  | ---                     | ---            | ---     | ---      | ---
 root   | read-write-1 | ---                     | ---            | ---     | ---      | ---
 user-3 | read-only-1  | ---                     | ---            | ---     | ---      | ---
 root   | read-write-1 | ---                     | ---            | ---     | ---      | ---
 user-5 | read-write-1 | 2018-08-21 11:04:16 (*) | OMERO.sessions | 10 min. | 141      | 2018-08-21 11:05:25
 root   | read-write-1 | ---                     | ---            | ---     | ---      | ---
 user-5 | read-write-1 | 2018-08-21 11:01:01     | OMERO.sessions | 10 min. | 83       | 2018-08-21 11:01:35
 user-3 | read-only-1  | ---                     | ---            | ---     | ---      | ---
(8 rows)
```
```
$ omero sessions list
 Server                 | User   | Group        | Session                              | Active    | Started
------------------------+--------+--------------+--------------------------------------+-----------+--------------------------
 eel.openmicroscopy.org | user-5 | read-write-1 | <SESSION-ID> | Logged in | Tue Aug 21 11:04:16 2018
(1 row)
```

**Review by manics — CHANGES_REQUESTED** [review]

File: omero/sysadmins/cli/usergroup.rst [hunk 8]
```diff
@@ -57,6 +57,33 @@ below:
...
+
+To deactivate a user, remove him/her from the system group ``user``.
+Use the command :program:`omero user leavegroup` and specify the ``user`` group as the target::
+
+    # Remove jsmith from group user
+    $ bin/omero user leavegroup user --name=jsmith
+
+To reactivate the user, add him/her back to the system group ``user`` i.e.::
+
+    $ bin/omero group adduser jsmith --name=user
```
Comment (joshmoore) [inline omero/sysadmins/cli/usergroup.rst]: I'd suggest the two commands being symmetric, i.e. leavegroup/joingroup or removeuser/adduser

File: omero/users/cli/sessions.rst (hunk 7 above)
Comment (joshmoore) [reply omero/users/cli/sessions.rst]: Agreed. `sessions list` is for the local work. `admin sessionlist` is what was deprecated in gh-5811

Comment (jburel) [reply omero/users/cli/sessions.rst]: no ref to ``sessionlist`` in the doc

File: omero/sysadmins/cli/usergroup.rst [hunk 9]
```diff
@@ -57,6 +57,33 @@ below:
...
+
+To deactivate a user, remove him/her from the system group ``user``.
+Use the command :program:`omero user leavegroup` and specify the ``user`` group as the target::
+
+    # Remove jsmith from group user
+    $ bin/omero user leavegroup user --name=jsmith
+
+To reactivate the user, add him/her back to the system group ``user`` i.e.::
+
+    $ bin/omero group joingroup jsmith --name=user
```
Comment (joshmoore) [inline omero/sysadmins/cli/usergroup.rst]: `bin/omero user joingroup user --name=jsmith`

File: omero/users/cli/sessions.rst [hunk 10]
```diff
@@ -109,7 +111,7 @@ Session keys can then be reused to switch between stored sessions using the
     $ bin/omero sessions login -k 22fccb8b-d04c-49ec-9d52-116a163728ca
     Server: [localhost]
     Joined session 1f800a16-1dc2-407a-8a85-fb44005306be (root@localhost:4064).
-    $ bin/omero sessions list
+    $ bin/omero sessions who
```
Comment (joshmoore) [inline omero/users/cli/sessions.rst]: This should still be list. A whole section on `sessions who` could either go here or on the sysadmin page.

Comment (jburel) [reply omero/users/cli/sessions.rst]: missed that one

File: omero/sysadmins/cli/usergroup.rst (hunk 9 above)
Comment (jburel) [reply omero/sysadmins/cli/usergroup.rst]: Both work but I see your point

Comment (joshmoore) [reply omero/sysadmins/cli/usergroup.rst]:
```
$ /opt/ome1/dist/bin/omero group joingroup
...
/opt/ome1/dist/bin/omero group: error: argument <subcommand>: invalid choice: 'joingroup'
```

**Comment by jburel** [comment]
@joshmoore done

File: omero/sysadmins/cli/usergroup.rst (hunk 9 above)
Comment (jburel) [reply omero/sysadmins/cli/usergroup.rst]: I see what I did. Looking back at my history
I ran the correct command and only partially update the doc
``group joingroup`` does not work

**Review by manics — APPROVED** [review]

File: omero/sysadmins/cli/usergroup.rst [hunk 11]
```diff
@@ -19,15 +19,15 @@ command::
 
 During the addition of the new user, you will need to specify the first and
 last name of the new user and their username as well as the groups the user
-belongs to. To add John Smith as a member of group 2 identified as jsmith,
+belongs to. To add John Smith as a member of the group named ``test-group`` identified as ``jsmith``,
```
Comment (pwalczysko) [inline omero/sysadmins/cli/usergroup.rst:22]: This still sounds just like if ``test-group`` was identified as ``jsmith``

I think that you need to shift the ``identified as jsmith`` bit to be just behind ``John Smith``.

Then ``To add John Smith, identified as ``jsmith`` as a member of the group named test-group...`` sound better to me

File: omero/sysadmins/cli/usergroup.rst [hunk 12]
```diff
@@ -19,15 +19,15 @@ command::
...
 enter::
 
-    $ bin/omero user add jsmith John Smith 2
+    $ bin/omero user add jsmith John Smith --group-name test-group
```
Comment (pwalczysko) [inline omero/sysadmins/cli/usergroup.rst:25]: I do not think you need the ``--group-name`` flag here.

For example

``bin/omero user add zpasta zubni pasta test-dom`` 
worked fine for me just now on eel, where ``test-dom`` is the name of a group on eel

```
bin/omero group info test-dom
Using session b81 (user-6@eel.openmicroscopy.org:4064). Idle timeout: 10 min. Current group: private-1
 id   | name     | perms  | ldap  | # of owners | # of members 
------+----------+--------+-------+-------------+--------------
 1603 | test-dom | rwr--- | False | 0           | 1            
(1 row)
```

File: omero/sysadmins/cli/usergroup.rst [hunk 13]
```diff
@@ -57,6 +57,33 @@ below:
...
+
+
+User editing
+^^^^^^^^^^^^
+
+Updating the details of a user e.g. the email address can be achieved using
+the :program:`omero obj update` command::
+
+    # Determine the ID of jsmith
+    $ bin/omero user info --user-name jsmith
```
Comment (pwalczysko) [inline omero/sysadmins/cli/usergroup.rst:82]: here, again the ``--user-name`` flag is not necessary.
See example below

```
bin/omero user info jsmith
Using session1 (user-6@eel.openmicroscopy.org:4064). Idle timeout: 10 min. Current group: private-1
 id   | login  | first name | last name | email | active | ldap  | admin | member of | owner of 
------+--------+------------+-----------+-------+--------+-------+-------+-----------+----------
 3102 | jsmith | john       | smith     |       |        | False |       | 1757      |          
(1 row)
```

File: omero/users/cli/sessions.rst [hunk 14]
```diff
@@ -74,7 +74,9 @@ Some of the options available to the :program:`omero login` command are:
 
 .. option:: -w PASSWORD, --password PASSWORD
 
-    Set the password to use for the connection.
+    Set the password to use for the connection. Since 5.4.1, the password can
+    be set using the :envvar:`OMERO_PASSWORD` environment variable. The variable
+    will be ignored if -w or --password is used.
```
Comment (pwalczysko) [inline omero/users/cli/sessions.rst:79]: I think the ``-w`` and ``--password`` should be formatted, probably in double inverted ticks. Now, especially the ``-w`` is a bit drowning in the surrounding text.

**Comment by pwalczysko** [comment]
Hmm, came late with my review, this is merged already :) Couple of comments above.

cc @joshmoore @jburel

File: omero/sysadmins/cli/usergroup.rst (hunk 13 above)
Comment (jburel) [reply omero/sysadmins/cli/usergroup.rst:82]: It was to make it more explicit

File: omero/sysadmins/cli/usergroup.rst (hunk 12 above)
Comment (jburel) [reply omero/sysadmins/cli/usergroup.rst:25]: mainly for clarity

File: omero/users/cli/sessions.rst (hunk 14 above)
Comment (jburel) [reply omero/users/cli/sessions.rst:79]: I can open a follow up PR to fix that

**Comment by jburel** [comment]
For the comments above, I opted for clarity vs "short" command since we have for example the  command ``user`` adding a user to the ``user`` group
and as we have seen previously the commands became nebulous for the reader (cf. import as feedback)

File: omero/users/cli/sessions.rst (hunk 14 above)
Comment (pwalczysko) [reply omero/users/cli/sessions.rst:79]: probably we can live with that now

**Comment by pwalczysko** [comment]
>For the comments above, I opted for clarity vs "short" command since we have for example the command user adding a user to the user group
and as we have seen previously the commands became nebulous for the reader (cf. import as feedback)

Okay, understood, ta

**Pull request merged**